org.scala-ide.sdt.core/src/org/scalaide/core/internal/jdt/model/ScalaClassFileProvider.scala

```diff
@@ -36,11 +53,8 @@ class ScalaClassFileProvider extends IClassFileProvider with HasLogger {
     scalaCF
   }
 
-  /** Return `true` if the classfile could be a Scala classfile.
-   *
-   *  @note This method caches the result of the first classfile read from a package fragment (usually a jar).
-   *        This heuristic might fail if a single jar mixes Java and Scala classfiles, and if the first classfile
-   *        is comes from Java, a plain Java classfile editor and icon would be used for all classfiles in that jar.
+  /**
+   * Returns `true` if the classfile could be a Scala classfile.
```

Comment:
Can you please describe the caching mechanism as it is now? I'm pretty sure this is non-obvious to anyone else but the two of us.